### ISSUE TITLE
Playlist only importing 100 songs

### ISSUE THREAD

**Issue by louocc** [Malopieds/InnerTune#136]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

1. Press account
2. Import a playlist over 100 songs
3. Go to library
4. See the playlist has only imported the first 100 songs

### Expected behavior

Import actual number of songs in playlist

### Actual behavior

Only imports 100

### Screenshots/Screen recordings

https://github.com/Malopieds/InnerTune/assets/158101536/af996a8e-b30c-479b-a208-5dc1135dc1fe

### Logs

n/a

### InnerTune version

0.5.7

### Android version

Android 14

**Comment by Irobrine** [comment]
Syncing seems to add the rest of the songs for me

**Comment by Malopieds** [comment]
> Syncing seems to add the rest of the songs for me

Yeah I've seen on your issue that you had 900+ songs, that why I thought it's odd.

> Import actual number of songs in playlist

Could you do three dots => select all and tell me the number of elements selected (my guess is youtube stop sending at 100). Maybe I can see if I there is a continuation with it

**Comment by Irobrine** [comment]
It says 946 elements selected for me

**Comment by Irobrine** [comment]
> Could you do three dots => select all and tell me the number of elements selected (my guess is youtube stop sending at 100). Maybe I can see if I there is a continuation with it

It says 946 elements selected

**Comment by louocc** [comment]
Turns out after reinstalling the app and restarting my phone and syncing again, it was able to find the rest of the songs after a few tries syncing. Thank you for your help.

**Comment by owenconnorz** [comment]
I tried this but still limited to 100 for me unfortunately

**Comment by Irobrine** [comment]
> I tried this but still limited to 100 for me unfortunately 

that's strange. what steps did you take? Are you logged in to your YT account?

**Comment by owenconnorz** [comment]
Yeah, I deleted the app, cleared data and cache, logged out, and then back in, and it's still broken for me, lol.

**Comment by Malopieds** [comment]
> Yeah, I deleted the app, cleared data and cache, logged out, and then back in, and it's still broken for me, lol.

Even after syncing it ?

**Comment by owenconnorz** [comment]
Ahh I deleted the app and installed again then Synced this time it worked as last time I clicked Synced it didn't do anything strange it doesn't just grab all the songs on the 1st import lol

**Comment by Malopieds** [comment]
> strange it doesn't just grab all the songs on the 1st import lol

Yes, it's because YouTube music doesn't send the whole playlist at first, they rather do it in a way of "continuation" so you have to fetch endpoint until you've reached the end. I will try to implement this whenever I have the time since there is a workaround as of now.
